src/main/webapp/wise5/components/summary/summary-authoring/summary-authoring.component.ts

```diff
@@ -1,56 +1,78 @@
 'use strict';
 
-import { Directive } from '@angular/core';
-import { EditComponentController } from '../../authoringTool/components/editComponentController';
-
-@Directive()
-class SummaryAuthoringController extends EditComponentController {
+import { Component } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+import { ProjectAssetService } from '../../../../site/src/app/services/projectAssetService';
+import { ComponentAuthoring } from '../../../authoringTool/components/component-authoring.component';
+import { ConfigService } from '../../../services/configService';
+import { NodeService } from '../../../services/nodeService';
+import { TeacherProjectService } from '../../../services/teacherProjectService';
+import { UtilService } from '../../../services/utilService';
+import { ConceptMapService } from '../../conceptMap/conceptMapService';
+import { MatchService } from '../../match/matchService';
+import { MultipleChoiceService } from '../../multipleChoice/multipleChoiceService';
+import { OpenResponseService } from '../../openResponse/openResponseService';
+import { SummaryService } from '../summaryService';
+
+@Component({
+  selector: 'summary-authoring.component',
+  templateUrl: 'summary-authoring.component.html',
+  styleUrls: ['summary-authoring.component.scss']
+})
+export class SummaryAuthoring extends ComponentAuthoring {
   isResponsesOptionAvailable: boolean = false;
   isHighlightCorrectAnswerAvailable: boolean = false;
   isPieChartAllowed: boolean = true;
-
-  static $inject = [
-    '$filter',
-    '$injector',
-    'ConfigService',
-    'NodeService',
-    'NotificationService',
-    'ProjectAssetService',
-    'ProjectService',
-    'SummaryService',
-    'UtilService'
-  ];
+  stepNodesDetails: string[];
+  componentTypeToService: {};
+  inputChange: Subject<string> = new Subject<string>();
+  inputChangeSubscription: Subscription;
 
   constructor(
-    protected $filter,
-    protected $injector,
-    protected ConfigService,
-    protected NodeService,
-    protected NotificationService,
-    protected ProjectAssetService,
-    protected ProjectService,
-    protected SummaryService,
-    protected UtilService
+    private ConceptMapService: ConceptMapService,
+    protected ConfigService: ConfigService,
+    protected NodeService: NodeService,
+    private MatchService: MatchService,
+    private MultipleChoiceService: MultipleChoiceService,
+    private OpenResponseService: OpenResponseService,
+    protected ProjectAssetService: ProjectAssetService,
+    protected ProjectService: TeacherProjectService,
+    private SummaryService: SummaryService,
+    private UtilService: UtilService
```

Comment:
Remove unnecessary params? See comment-1 below.
```suggestion
    protected ConfigService: ConfigService,
    protected NodeService: NodeService,
    protected ProjectAssetService: ProjectAssetService,
    protected ProjectService: TeacherProjectService,
    private SummaryService: SummaryService,
    private UtilService: UtilService
```